edb/server/protocol/auth_ext/http.py

```diff
@@ -1317,6 +1390,37 @@ async def _try_verify_email(
                 " in our system, or it might already be verified."
             )
 
+    def _is_url_allowed(self, url: str) -> bool:
+        allowed_urls = util.get_config(
+            self.db,
+            "ext::auth::AuthConfig::allowed_redirect_urls",
+            frozenset,
+        )
+        allowed_urls = cast(Set[str], allowed_urls)
+        parsed_url = urllib.parse.urlparse(url)
+
+        for allowed_url in allowed_urls:
```

Comment:
Why not just do string prefix comparison as `any(url.startswith(allowed) for allowed in allowed_urls))`?  Also, should this be done case-insensitively?

Reply:
I was trying to match GitHub's matching algorithm which does this domain-minus-subdomain matching.

Reply:
@elprans I ended up just going with a `for` loop with an early `return` since that seems _slightly_ better than iterating the whole list? Thoughts?

Reply:
The `any`+generator fragment that Elvis suggested will short circuit also, since `any` will return as soon as it sees a true element, and so the rest of the generator won't execute.
Though probably your version is still faster because it doesn't need to deal with a generator.